controllers/haproxyloadbalancer_controller.go

```diff
@@ -778,3 +798,29 @@ func (r haproxylbReconciler) controlPlaneMachineToHAProxyLoadBalancer(o handler.
 		},
 	}}
 }
+
+func (r *haproxylbReconciler) reconcileClusterToHAProxyLoadBalancers(a handler.MapObject) []reconcile.Request {
+	requests := []reconcile.Request{}
+	lbs := &infrav1.HAProxyLoadBalancerList{}
+	err := r.Client.List(goctx.Background(),
+		lbs,
+		ctrlclient.InNamespace(a.Meta.GetNamespace()),
+		ctrlclient.MatchingLabels(
+			map[string]string{
+				clusterv1.ClusterLabelName: a.Meta.GetName(),
+			},
+		))
+	if err != nil {
+		return requests
+	}
+	for _, lb := range lbs.Items {
+		r := reconcile.Request{
+			NamespacedName: apitypes.NamespacedName{
```

Comment:
No need to change but just as an FYI for the future, CAPI now has `util.ObjectKey()` that you can use to get this, instead of typing it out.

Reply:
I'll add this to things to file out of this PR with @detiber 's comments